robolectric/src/main/java/org/robolectric/RobolectricTestRunner.java

```diff
@@ -487,6 +493,32 @@ protected void configureShadows(SdkEnvironment sdkEnvironment, Config config) {
     injectEnvironment(sdkEnvironment.getRobolectricClassLoader(), classHandler, sdkEnvironment.getShadowInvalidator());
   }
 
+  @VisibleForTesting
+  void resetUserShadows(Config config) throws InvocationTargetException, IllegalAccessException {
+    Class<?>[] shadows = config.shadows();
+    for (Class c : shadows) {
```

Comment:
Should probably check that the shadow class is applicable to the current SDK level (`@Config(minSdk/maxSdk)`).